File: src/main/java/com/example/the_machine/dto/ArtifactDto.java
package com.example.the_machine.dto;

import com.example.the_machine.db.ArtifactEntity;
import com.fasterxml.jackson.databind.JsonNode;
import java.time.Instant;
import java.util.UUID;

public record ArtifactDto(
    UUID id,
    UUID projectId,
    UUID runId,
    ArtifactEntity.Kind kind,
    ArtifactEntity.Status status,
    String title,
    String mime,
    String url,                  // computed if file-backed
    JsonNode content,            // structured outputs
    JsonNode metadata,           // dims, duration, etc.
    Instant createdAt,
    UUID supersededById
) {

}

File: src/main/kotlin/com/example/the_machine/dto/ArtifactDto.kt
package com.example.the_machine.dto

import com.example.the_machine.domain.ArtifactEntity
import kotlinx.serialization.json.JsonElement
import java.time.Instant
import java.util.*

data class ArtifactDto(
  val id: UUID,
  val threadId: UUID,
  val runId: UUID?,
  val kind: ArtifactEntity.Kind,
  val stage: ArtifactEntity.Stage,
  val status: ArtifactEntity.Status,
  val title: String?,
  val mime: String?,
  val url: String?,                  // computed if file-backed
  val content: JsonElement?,                 // structured outputs
  val metadata: JsonElement?,                // dims, duration, etc.
  val createdAt: Instant?,
  val supersededById: UUID?
)